app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/LectureViewDrawer.kt

```diff
@@ -0,0 +1,237 @@
+            lp.bottomMargin = params.bottomMargin
+            lp.topMargin = params.topMargin
+            event.layoutParams = lp
+
+            updateEventView(event, lecture)
+        }
+    }
+
+    fun setLectureBackground(event: Lecture, eventView: View) {
+        val context = eventView.context
+        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
+        val defaultValue = resources.getBoolean(R.bool.preferences_alternative_highlight_enabled_default_value)
+        val alternativeHighlightingIsEnabled = prefs.getBoolean(
+                BundleKeys.PREFS_ALTERNATIVE_HIGHLIGHT, defaultValue)
+        val eventIsFavored = event.highlight
+        @ColorRes val backgroundColorResId: Int
+        backgroundColorResId = if (eventIsFavored) {
+            val colorResId = trackNameBackgroundColorHighlightPairs[event.track]
+            colorResId ?: R.color.event_border_highlight
+        } else {
+            val colorResId = trackNameBackgroundColorDefaultPairs[event.track]
+            colorResId ?: R.color.event_border_default
+        }
+        @ColorInt val backgroundColor = ContextCompat.getColor(context, backgroundColorResId)
+        val eventDrawable: EventDrawable
+        eventDrawable = if (eventIsFavored && alternativeHighlightingIsEnabled) {
+            EventDrawable(
+                    backgroundColor,
+                    eventDrawableCornerRadius.toFloat(),
+                    eventDrawableRippleColor,
+                    eventDrawableStrokeColor,
+                    eventDrawableStrokeWidth.toFloat())
+        } else {
+            EventDrawable(
+                    backgroundColor,
+                    eventDrawableCornerRadius.toFloat(),
+                    eventDrawableRippleColor)
+        }
+        eventDrawable.setLayerInset(EventDrawable.BACKGROUND_LAYER_INDEX,
+                eventDrawableInsetLeft,
+                eventDrawableInsetTop,
+                eventDrawableInsetRight,
+                0)
+        eventDrawable.setLayerInset(EventDrawable.STROKE_LAYER_INDEX,
+                eventDrawableInsetLeft,
+                eventDrawableInsetTop,
+                eventDrawableInsetRight,
+                0)
+        eventView.setBackgroundDrawable(eventDrawable)
+        val padding = eventPadding
+        eventView.setPadding(padding, padding, padding, padding)
+    }
+
+    fun setLectureTextColor(lecture: Lecture, view: View) {
+        val title = view.findViewById<TextView>(R.id.event_title)
+        val subtitle = view.findViewById<TextView>(R.id.event_subtitle)
+        val speakers = view.findViewById<TextView>(R.id.event_speakers)
+        val colorResId = if (lecture.highlight) R.color.event_title_highlight else R.color.event_title
+        val textColor = ContextCompat.getColor(view.context, colorResId)
+        title.setTextColor(textColor)
+        subtitle.setTextColor(textColor)
+        speakers.setTextColor(textColor)
+    }
```

Comment:
Same as above, see https://github.com/EventFahrplan/EventFahrplan/pull/256#discussion_r404126003

Reply:
This is not about rewriting logic. It is the same as the Java code just written as idiomatic Kotlin. When converting code from Java to Kotlin do not trust the auto-converter of Android Studio to do the best job.

Reply:
I agree, I can be static, at least for now :). Since the method implementation should be open for changes and the method API closed for modifications, I leave the lecture parameter as is. If at some point other properties of lecture are required, the signature stays the same. Also Extension methods are somehow verbose, since an extension method on View class would make this method available on any view instance, and I would like to prevent that.

Reply:
The extension function could be limited to `TextView`. I hope that the new _view binding_ feature introduces more specific types for this kind of pattern. I prefer a object-oriented style of forming method signatures. - I also agree that it pollutes the `View` namespace somewhat - at least when the class is imported. I don't insist on the extension function.
With regards to improving the auto-conversion I would still argue that it's worth it.

Reply:
I decided to leave it as is.

Reply:
Updated in e75b82f43de088e5edbc87f9a6722a8584c0dacb. Thx.